src/utils/debounce.ts

```diff
@@ -0,0 +1,11 @@
+export const debounce = (callback: (...args: any[]) => void, wait: number): ((...args: any[]) => void) => {
+  let timeoutId: number
+
+  return (...args: any[]) => {
+    window.clearTimeout(timeoutId)
+    timeoutId = window.setTimeout(() => {
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      callback(...args)
+    }, wait)
+  }
+}
```

Comment:
We had a sync discussion about including this instead of something like @lodash/debounce. This change makes it a little bit better on the eyes.

```suggestion
export const debounce = (callback: (...args: unknown[]) => void, wait: number): ((...args: unknown[]) => void) => {
  let timeoutId: number

  return (...args: unknown[]) => {
    window.clearTimeout(timeoutId)
    timeoutId = window.setTimeout(() => {
      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
      callback(...args)
    }, wait)
  }
}
```

Reply:
having unkown as the first type caused type errors at implementation. 
i've [made some changes](https://github.com/mParticle/aquarium/blob/1dcdf0e229ea1e3a2bc8ef3d6cae09d290bc50cb/src/utils/utils.ts#L15-L14) and moved it into a utils file in the followup pr